src/adapter/src/catalog.rs

```diff
@@ -3515,6 +3593,51 @@ impl From<PlanContext> for SerializedPlanContext {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum SerializedComputeInstanceReplicaConfig {
+    Remote {
+        addrs: BTreeSet<String>,
+    },
+    Managed {
+        size: String,
+        availability_zone: Option<String>,
+    },
+}
```

Comment:
Not sure I understand the intent -- how does this differ from `ComputeInstanceReplicaConfig` ? Can you leave a comment somewhere now that there are three different types (serialized, concrete, and vanilla) explaining the conceptual purpose of each?